src/test/java/io/nats/compatibility/ObjectStoreCommand.java

```diff
@@ -0,0 +1,280 @@
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchWatcher ww = new WatchWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(ww)) {
+                ww.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(ww.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = result + "," + oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    private void doWatchUpdate() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchUpdatesWatcher wuw = new WatchUpdatesWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(wuw)) {
+                wuw.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(wuw.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchUpdatesWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    protected void _respondDigest(ObjectInfo oi) {
+        String digest = oi.getDigest();
+        Log.info("RESPOND " + subject + " digest " + digest);
+        byte[] payload = Base64.getUrlDecoder().decode(digest.substring(8));
+        nc.publish(replyTo, payload);
+    }
+
+    private void _putObject(String objectName, String description) {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            ObjectMeta meta = ObjectMeta.builder(objectName).description(description).build();
+            InputStream inputStream = Utility.getFileAsInputStream("nats-server.zip");
+            os.put(meta, inputStream);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void _createBucket() {
+        ObjectStoreManagement osm;
+        try {
+            osm = nc.objectStoreManagement();
+        }
+        catch (IOException e) {
+            handleException(e);
+            return;
+        }
+
+        ObjectStoreConfiguration osc = extractObjectStoreConfiguration();
+        try {
+            osm.delete(osc.getBucketName());
```

Comment:
You don't need that here - compat tester handles the cleanup for you.

Reply:
Except if you kill the test in the middle, the bucket is sometimes left around. This caused much pain, so once I figured it out, I left it in there.